federation/pkg/federation-controller/service/service_controller.go

```diff
@@ -173,6 +175,34 @@ func NewServiceController(client fedclientset.Interface) *ServiceController {
 		},
 	)
 
+	// Federated informer on endpoints in members of federation to know backing healthy endpoints
+	sc.endpointFederatedInformer = util.NewFederatedInformer(
+		client,
+		func(cluster *fedv1.Cluster, targetClient kubeclientset.Interface) (
+			cache.Store, cache.ControllerInterface) {
+			return cache.NewInformer(
+				&cache.ListWatch{
+					ListFunc: func(options apiv1.ListOptions) (runtime.Object, error) {
+						return targetClient.Core().Endpoints(apiv1.NamespaceAll).List(options)
+					},
+					WatchFunc: func(options apiv1.ListOptions) (watch.Interface, error) {
+						return targetClient.Core().Endpoints(apiv1.NamespaceAll).Watch(options)
+					},
+				},
+				&apiv1.Endpoints{},
+				time.Minute*10, // rechecks all endpoints on interval
```

Comment:
agreed, sorry for the mistake. I was double ensuring if at all we miss an watch event on update. but i think its not the case and probable we can go with NoResyncPeriodFunc() as followed by other controllers.